### DISCUSSION TITLE
eslint-plugin-vue vs vuetify

### DISCUSSION THREAD

**Discussion by icleolion** [vuetifyjs/vuetify#11486] [General] [answered]
Don't suppose anyone has noticed vuetify's usage of v-slot modifiers is invalid according to eslint-plugin-vue's valid-v-slot rule?
![image](https://user-images.githubusercontent.com/47498402/82851832-04098700-9ef9-11ea-8e56-c712411920f4.png)

Vuetify documentation example -> https://codepen.io/pen/?&editable=true&editors=101=https%3A%2F%2Fvuetifyjs.com%2Fen%2Fcomponents%2Fdata-tables%2F
eslint-plugin-vue rule documentation -> https://eslint.vuejs.org/rules/valid-v-slot.html

Also it isn't happy about the below either, but this i'm not too fussed about:
![image](https://user-images.githubusercontent.com/47498402/82851807-ef2cf380-9ef8-11ea-8527-a845e1387b5b.png)

So who is right and who is wrong, or are you both right and I shouldn't bother creating an issue about it on either of your github repositories? Discuss... :wink:

**Comment by KaelWD** [comment, marked as answer]
Your codepen link is blank, but I'm assuming it's about `item.foo` slots. eslint-plugin-vue is *technically* correct, but v-slot is a compiler directive and actually ignores modifiers - `v-slot:foo.bar` is compiled to `scopedSlots: { 'foo.bar' () { ...` instead of `directives: [{ name: 'slot', arg: 'foo', modifiers: { bar: true } }]`

So eslint-plugin-vue is actually wrong here, because this convention does work as expected.

**Comment by icleolion** [comment]
Bugger would help if I saved the codepen after pushing it out of the documentation!
https://codepen.io/icleolion/pen/ExVJGbv

But anyways, you were right about what I was getting at.  I'll log the issue with eslint-plugin-vue then.  See if they have a counter argument and send me back your way 😅 

EDIT: issue has been logged https://github.com/vuejs/eslint-plugin-vue/issues/1165

**Comment by icleolion** [comment]
I see @KaelWD is taking this up with the Vue.js team directly now, so I consider this discussion complete as far as Vue 2 is concerned.  We shall see how things play out for Vue 3. 🤞 

https://github.com/vuejs/vue-next/issues/1241
